src/databricks/labs/ucx/source_code/queries.py

```diff
@@ -43,41 +43,56 @@ def schema(self):
         return self._session_state.schema
 
     def lint(self, code: str) -> Iterable[Advice]:
-        for statement in sqlglot.parse(code, read='databricks'):
-            if not statement:
-                continue
-            for table in statement.find_all(Table):
-                if isinstance(statement, Use):
-                    # Sqlglot captures the database name in the Use statement as a Table, with
-                    # the schema  as the table name.
-                    self._session_state.schema = table.name
-                    continue
-                if isinstance(statement, Create) and statement.kind == "SCHEMA":
-                    # Sqlglot captures the schema name in the Create statement as a Table, with
-                    # the schema  as the db name.
-                    self._session_state.schema = table.db
+        try:
+            statements = parse_sql(code, read='databricks')
+            for statement in statements:
+                if not statement:
                     continue
+                yield from self._lint_statement(statement)
+        except SqlParseError:
```

Comment:
At the moment for people running into this there's no way for them to figure out why something is failing to parse; at the very least we should log the cause of the SQL error. To keep the noise down I'd be okay with `DEBUG` (and `exc_info=True`); it's really about ensuring we can get the information if we need it without having to reproduce it manually ourselves.

Reply:
done